Components/Terrain/src/OgreTerrainGroup.cpp

```diff
@@ -693,25 +696,23 @@ namespace Ogre
     //---------------------------------------------------------------------
     void TerrainGroup::handleResponse(const WorkQueue::Response* res, const WorkQueue* srcQ)
     {
+        // Data was already deleted so nothing we can do anymore.
+        if (res->getRequest()->getAborted())
+            return;
+
         // No response data, just request
         LoadRequest lreq = any_cast<LoadRequest>(res->getRequest()->getData());
-        --LoadRequest::loadingTaskNum;
 
-        TerrainLoadRequestMap::iterator it = mTerrainLoadRequests.find(lreq.slot->instance);
+        TerrainPrepareRequestMap::iterator it = mTerrainPrepareRequests.find(lreq.slot);
 
-        assert(it != mTerrainLoadRequests.end());
-
-        if (it != mTerrainLoadRequests.end())
+        // This slot was scheduled to be deleted.
+        if (it == mTerrainPrepareRequests.end())
         {
-            bool isRemoved = it->second;
-            mTerrainLoadRequests.erase(it);
-
-            // Instance was scheduled for removal while it was processed in another thread.
-            if (isRemoved)
-            {
-                freeTerrainSlotInstance(lreq.slot);
-                return;
-            }
+            freeTerrainSlotInstance(lreq.slot);
```

Comment:
There should be return; after the free here. I'll try if this fixes the neighbour issue.

Reply:
Ok that did not fix the issue, unfortunately.